### PR TITLE
Add `update-pr-from-base-branch` feature

### PR DESCRIPTION
**Pull request by bfred-it**
Closes https://github.com/sindresorhus/refined-github/issues/1236

- It works only when the branches are on the same repo, because of API limitations
- It works even if `master` was force-pushed
- The name is simplified to `master` because people _get it_ but it works on any branch

# Tasks

- [x] Merge with master, it needs a description field because of #2030

<details><summary>Edit: all resolved</summary>
- [x] Make less prominent, this shouldn't look like a necessary step.

    **HELP:** _Where/how should we show this button? Does it make sense there? This is what I got so far, which doesn't look right (warning icon, too prominent)_

- [x] The button might not appear sometimes; I'll fix this once we figure out where to place it 👆
- [x] Only show if necessary, like when a branch is stale or when GH already shows it 

    **HELP:** _how do we find out whether a branch is stale?_ Edit: maybe this https://developer.github.com/v3/repos/commits/#compare-two-commits

- [x] Handle errors. I'll probably change the API to just `ignoreHTTPStatus: false` instead of `accept404`. API reference: https://developer.github.com/v3/repos/merging/
- [x] Handle conflict (i.e. console error) with `add-co-authors` (fixed in #2019)
- [x] Handle possible caching issue after merging (the button reappears)
- [-] Update readme

cc @rarkins @sindresorhus 
</details>

# Test

Test it on any PR. It should depend on:

- User has permission to merge or not
- PR can be merged or not (e.g. conflicts or reviews required)
- PR type (draft or normal)

~~Outdated screenshot https://user-images.githubusercontent.com/1402241/56434750-842de800-6308-11e9-9092-75cd703d4414.png~~

### CODE REVIEW

**Comment by bfred-it** [comment]
The commit above is done with this feature 🎉 https://github.com/sindresorhus/refined-github/pull/1950/commits/371960a3d7b7dfc3e12714e102c36342e5b61889

**Comment by sindresorhus** [comment]
> Only show if necessary, like when a branch is stale or when GH already shows it

Yes, only show it if the PR branch behind `master`.

> Make less prominent, this shouldn't look like a necessary step.

Agreed. It's only a helpful utility, not a required operation. Not sure where to place it though. Maybe a small link here: 
Screen Shot 2019-04-26 at 13 09 45

File: source/features/update-pr-from-master.tsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
+import React from 'dom-chef';
+import select from 'select-dom';
+import features from '../libs/features';
+import * as api from '../libs/api';
+import * as icons from '../libs/icons';
+
+export async function mergeBranches(repoUrl: string, base: string, head: string) {
+	const result = await api.v3(`repos/${repoUrl}/merges`, {
```
Comment (sindresorhus) [inline source/features/update-pr-from-master.tsx]: The `v4` API doesn't support this?

**Comment by rarkins** [comment]
Re: default vs master terminology. Isn't it technically a third option - i.e. whatever is the "base" branch for the PR?

For example if my default branch is `master` but I have another branch called something like `v3` which is for the next major release, then I may have PRs targeting `v3` as the base branch and therefore to sync it with default/master would be incorrect. In such situations we should probably update it from `v3` or hide the option if it is not possible.

**Comment by bfred-it** [comment]
I used `master` also because we have another feature named after it, but `base` is actually more correct because that's what this feature looks for.

File: source/features/update-pr-from-master.tsx (hunk 1 above)
Comment (bfred-it) [reply source/features/update-pr-from-master.tsx]: I think not, and even then the REST query is easier to build

**Comment by bfred-it** [comment]
conflict

**Comment by bfred-it** [comment]
Even better, link to `/conflicts` when there are conflicts in Draft PRs

**Comment by sindresorhus** [comment]
Seems to work, however:

> Only show if necessary, like when a branch is stale or when GH already shows it

This doesn't work correctly. I made a PR (https://github.com/sindresorhus/sindre-playground/pull/4/), then updated a file in master, and reloaded the PR, got the "Update branch" button, clicked it, got "Updating branch...", then it disappeared, and then reappeared again, if I click it now, it just switches to "Updating branch...", but nothing happens. If I reload the page 1 minute later, the button no longer shows.

I have a feeling that this is caused by the API taking some time to update its status or something. Still, it's not a very good experience for users.

**Comment by bfred-it** [comment]
> got the "Update branch" button, clicked it, got "Updating branch...", then it disappeared, and then reappeared again, if I click it now, it just switches to "Updating branch...", but nothing happens. If I reload the page 1 minute later, the button no longer shows.

Yeah that’s the second-last to-do on my list, I think I’ll just have to disable the feature after the merge because it’s unlikely to be needed right away

**Comment by bfred-it** [comment]
I think this is good to go

Screenshot 2019-05-15 at 23 53 10

**Comment by sindresorhus** [comment]
Looks good now. The only problem is that when you click "Update branch", then it updates, and you refresh the page, the button is there again, as the API is slow to update... I don't think we can do anything about that though.

**Comment by rarkins** [comment]
Thanks all

**Pull request merged**